### PR TITLE
Reorganize root README and add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by austinborn**
## Description
 Reorganize `README.md` and add `CONTRIBUTING.md` for info on how to contribute to the repo.

## Changes

- Reorder sections
- Condense Advanced Features subsections
- Simplify table of contents
- Minor grammar and word fixes
- Change code block text types in some places
- Move certain sections to `CONTRIBUTE.md` with its own table of contents

## Steps to Test

(No tests required)

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  1. Full-featured "includes" array (in the format of [presetIncludes.json](packages/core/bootstrap/src/lib/external-adapter/overrides/presetIncludes.json))
+  2. Pre-set includes from the EA (set in [presetIncludes.json](packages/core/bootstrap/src/lib/external-adapter/overrides/presetIncludes.json))
+  3. String array as "includes"
+
+## Mock Integration Testing
+
+We use [Nock](https://github.com/nock/nock) for intercepting HTTP requests in integration tests and returning mock data.
+The [recording](https://github.com/nock/nock#recording) functionality of nock is used when first writing the test to automatically generate accurate fixture data.
+
+For example, take a look at the [synth-index] (./packages/composites/synth-index/test/integration/adapter.test.ts) test to see it in usage. When the `RECORD` environment variable is truthy, nock will proxy HTTP requests and generate fixture data that can be used to contruct the integration test.
```
Comment (boxhock) [inline CONTRIBUTING.md:93]:
```suggestion
For example, take a look at the [synth-index](./packages/composites/synth-index/test/integration/adapter.test.ts) test to see it in usage. When the `RECORD` environment variable is truthy, nock will proxy HTTP requests and generate fixture data that can be used to contruct the integration test.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+The follow steps is the general pattern for writing an integration test.
+
+1. Setup nock to record HTTP requests, see the [synth-index] (./packages/composites/synth-index/test/integration/adapter.test.ts) test for a code sample.
```
Comment (boxhock) [inline CONTRIBUTING.md:97]:
```suggestion
1. Setup nock to record HTTP requests, see the [synth-index](./packages/composites/synth-index/test/integration/adapter.test.ts) test for a code sample.
```

**Review by boxhock — CHANGES_REQUESTED** [review]
Looks good besides the broken link here! Thanks for doing this :grinning:

**Review by justinkaseman — DISMISSED** [review]
LGTM

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+Thank you for your interest in improving the Chainlink External Adapter codebase! The steps below support development of original adapters, but you are more than welcome to contribute to existing adapters as well. When opening a PR, please invite `smartcontractkit/solutions-engineering` to review the changes.
+
+## Table of Contents
+
+1. [Creating A New Adapter](#Creating-A-New-Adapter)
+2. [Input](#Input)
+3. [Output](#Output)
+4. [Common Patterns](#Common-Patterns)
+5. [Mock Integration Testing](#Mock-Integration-Testing)
+6. [Soak Testing (Chainlink Team only)](<#Soak-Testing-(Chainlink-Team-only)>)
```
Comment (austinborn) [inline CONTRIBUTING.md:12]:
```suggestion
6. [Soak Testing (Chainlink Labs)](<#Soak-Testing-(Chainlink-Labs)>)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+The follow steps is the general pattern for writing an integration test.
+
+1. Setup nock to record HTTP requests, see the [synth-index](./packages/composites/synth-index/test/integration/adapter.test.ts) test for a code sample.
+2. Run the test, using live API endpoints for the external adapter under test to hit, with nock recording on (`export RECORD=true`).
+3. Using the generated fixture data from step 2, write a `fixtures.ts` file to return the mock data instead.
+4. Run the tests again with nock recording disabled (`unset RECORD`). API requests should now be intercepted and mocked using the fixture data. Be sure to run tests with the `--updateSnapshot` flags to update the integration snapshot if necessary.
+
+For more information on Jest, see the [Jest docs](https://jestjs.io/docs/cli).
+
+## Soak Testing (Chainlink Team only)
```
Comment (austinborn) [inline CONTRIBUTING.md:105]:
```suggestion
## Soak Testing (Chainlink Labs)
```

**Review by boxhock — APPROVED** [review]

**Pull request merged**